### ISSUE TITLE
The exclude-results do not work correctly

### ISSUE THREAD

**Issue by ansidorov** [Checkmarx/kics#2162]
### Expected Behavior
There should be no check in the result of the scan `Pod Is Not Targeted By A Proper Network Policy` with `query_id: 0401f71b-9c1e-4821-ab15-a955caa621be`

### Steps to Reproduce the Problem

```
# git clone https://github.com/rbaumgar/statefulsets.git
# cd statefulsets
# docker run -v $(pwd):/path  --rm -ti checkmarx/kics:latest scan -p "/path"  -o "/path/result.json"
Scanning with Keeping Infrastructure as Code Secure development

Executing queries: [========== 100.0% ==========]
Files scanned: 1
Parsed files: 3
Queries loaded: 1244
Queries failed to execute: 0
------------------------------------
Service Account Token Automount Not Disabled, Severity: MEDIUM, Results: 1
        /path/zookeeper.yaml:48
StatefulSet Requests Storage, Severity: LOW, Results: 1
        /path/zookeeper.yaml:126
StatefulSet Without PodDisruptionBudget, Severity: LOW, Results: 1
        /path/zookeeper.yaml:37
Object Is Using A Deprecated API Version, Severity: HIGH, Results: 1
        /path/zookeeper.yaml:31
Seccomp Profile Is Not Configured, Severity: MEDIUM, Results: 1
        /path/zookeeper.yaml:45
Service Does Not Target Pod, Severity: LOW, Results: 2
        /path/zookeeper.yaml:15
        /path/zookeeper.yaml:28
Using Default Namespace, Severity: MEDIUM, Results: 3
        /path/zookeeper.yaml:5
        /path/zookeeper.yaml:21
        /path/zookeeper.yaml:34
Pod Is Not Targeted By A Proper Network Policy, Severity: MEDIUM, Results: 2
        /path/zookeeper.yaml:7
        /path/zookeeper.yaml:23

Results Summary:
HIGH: 1
MEDIUM: 7
LOW: 4
INFO: 0
TOTAL: 12

Scan duration: 4.5974483s
```
Let's make sure there is such a check as a result:
```
# cat result.json| jq '.queries[] | select(.query_id == "0401f71b-9c1e-4821-ab15-a955caa621be")'
{
  "query_name": "Pod Is Not Targeted By A Proper Network Policy",
  "query_id": "0401f71b-9c1e-4821-ab15-a955caa621be",
  "category": "Networking",
  "description": "Check if any pod is not being targeted by a proper network policy.",
  "severity": "MEDIUM",
  "platform": "Kubernetes",
  "files": [
    {
      "file_name": "/path/zookeeper.yaml",
      "similarity_id": "eb7ba423b2e870b5576dddc1b56f77eaf131f3872178c1e2228411d8aad7d92f",
      "line": 7,
      "issue_type": "MissingAttribute",
      "search_key": "metadata.name=zk-hs.labels.app",
      "search_value": "",
      "expected_value": "'metadata.labels.app' is targeted by a network policy",
      "actual_value": "'metadata.labels.app' is not targeted by a network policy",
      "value": null
    },
    {
      "file_name": "/path/zookeeper.yaml",
      "similarity_id": "e0769e8e5ba189cc87bb37d02a7913c6d97e23c8d7c8423472aac750b060d5bd",
      "line": 23,
      "issue_type": "MissingAttribute",
      "search_key": "metadata.name=zk-cs.labels.app",
      "search_value": "",
      "expected_value": "'metadata.labels.app' is targeted by a network policy",
      "actual_value": "'metadata.labels.app' is not targeted by a network policy",
      "value": null
    }
  ]
}
```
Delete the file with the result:
```
# rm result.json
```

Let's run the check again with the `-x' parameter:
```
# docker run -v $(pwd):/path  --rm -ti checkmarx/kics:latest scan -p "/path" -x "0401f71b-9c1e-4821-ab15-a955caa621be" -o "/path/result.json"
Scanning with Keeping Infrastructure as Code Secure development

Executing queries: [========== 100.0% ==========]
Files scanned: 1
Parsed files: 3
Queries loaded: 1244
Queries failed to execute: 0
------------------------------------
StatefulSet Requests Storage, Severity: LOW, Results: 1
        /path/zookeeper.yaml:126
StatefulSet Without PodDisruptionBudget, Severity: LOW, Results: 1
        /path/zookeeper.yaml:37
Seccomp Profile Is Not Configured, Severity: MEDIUM, Results: 1
        /path/zookeeper.yaml:45
Service Account Token Automount Not Disabled, Severity: MEDIUM, Results: 1
        /path/zookeeper.yaml:48
Service Does Not Target Pod, Severity: LOW, Results: 2
        /path/zookeeper.yaml:15
        /path/zookeeper.yaml:28
Using Default Namespace, Severity: MEDIUM, Results: 3
        /path/zookeeper.yaml:5
        /path/zookeeper.yaml:21
        /path/zookeeper.yaml:34
Object Is Using A Deprecated API Version, Severity: HIGH, Results: 1
        /path/zookeeper.yaml:31
Pod Is Not Targeted By A Proper Network Policy, Severity: MEDIUM, Results: 2
        /path/zookeeper.yaml:7
        /path/zookeeper.yaml:23

Results Summary:
HIGH: 1
MEDIUM: 7
LOW: 4
INFO: 0
TOTAL: 12

Scan duration: 4.7197274s
```
Make sure the result file has this query in it:
```
# cat result.json| jq '.queries[] | select(.query_id == "0401f71b-9c1e-4821-ab15-a955caa621be")'
{
  "query_name": "Pod Is Not Targeted By A Proper Network Policy",
  "query_id": "0401f71b-9c1e-4821-ab15-a955caa621be",
  "category": "Networking",
  "description": "Check if any pod is not being targeted by a proper network policy.",
  "severity": "MEDIUM",
  "platform": "Kubernetes",
  "files": [
    {
      "file_name": "/path/zookeeper.yaml",
      "similarity_id": "eb7ba423b2e870b5576dddc1b56f77eaf131f3872178c1e2228411d8aad7d92f",
      "line": 7,
      "issue_type": "MissingAttribute",
      "search_key": "metadata.name=zk-hs.labels.app",
      "search_value": "",
      "expected_value": "'metadata.labels.app' is targeted by a network policy",
      "actual_value": "'metadata.labels.app' is not targeted by a network policy",
      "value": null
    },
    {
      "file_name": "/path/zookeeper.yaml",
      "similarity_id": "e0769e8e5ba189cc87bb37d02a7913c6d97e23c8d7c8423472aac750b060d5bd",
      "line": 23,
      "issue_type": "MissingAttribute",
      "search_key": "metadata.name=zk-cs.labels.app",
      "search_value": "",
      "expected_value": "'metadata.labels.app' is targeted by a network policy",
      "actual_value": "'metadata.labels.app' is not targeted by a network policy",
      "value": null
    }
  ]
}
```

### Specifications

-   Version: 
```
docker run -v $(pwd):/path  --rm -ti checkmarx/kics:latest version
Keeping Infrastructure as Code Secure development
```
-   Platform: docker
-   Subsystem:

**Comment by ruigomescx** [comment]
Hi @ansidorov. In this case, our -x flag is designed to exclude individual results. To do this you can pass the similarity_id (not the queryid) field of the result, to the -x flag, and it will exclude the result.

However, we can also try to support this exclusion of results by query. Would this be helpful?

**Comment by ansidorov** [comment]
Yes, it would be very cool if it were possible to exclude individual query and entire query categories.

**Comment by ruigomescx** [comment]
Hi @ansidorov. Now with the --exclude-queries flag, you can pass individual queries that you want to exclude. This is available from version 1.2.0. Thank you for the feedback!
